tests/integration/golang/aim/run/get_runs_active_test.go

```diff
@@ -0,0 +1,116 @@
+//go:build integration
+
+package run
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/G-Research/fasttrackml/pkg/api/aim/encoding"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/dao/models"
+	"github.com/G-Research/fasttrackml/tests/integration/golang/fixtures"
+	"github.com/G-Research/fasttrackml/tests/integration/golang/helpers"
+)
+
+type GetRunsActiveTestSuite struct {
+	suite.Suite
+	client             *helpers.HttpClient
+	runFixtures        *fixtures.RunFixtures
+	experimentFixtures *fixtures.ExperimentFixtures
+	runs               []*models.Run
+}
+
+func TestGetRunsActiveTestSuite(t *testing.T) {
+	suite.Run(t, new(GetRunsActiveTestSuite))
+}
+
+func (s *GetRunsActiveTestSuite) SetupTest() {
+	s.client = helpers.NewAimApiClient(helpers.GetServiceUri())
+
+	runFixtures, err := fixtures.NewRunFixtures(helpers.GetDatabaseUri())
+	assert.Nil(s.T(), err)
+	s.runFixtures = runFixtures
+
+	expFixtures, err := fixtures.NewExperimentFixtures(helpers.GetDatabaseUri())
+	assert.Nil(s.T(), err)
+	s.experimentFixtures = expFixtures
+
+	exp := &models.Experiment{
+		Name:           uuid.New().String(),
+		LifecycleStage: models.LifecycleStageActive,
+	}
+	_, err = s.experimentFixtures.CreateExperiment(context.Background(), exp)
+	assert.Nil(s.T(), err)
+
+	s.runs, err = s.runFixtures.CreateRuns(context.Background(), exp, 3)
+	assert.Nil(s.T(), err)
+
+	// set 3rd run to status = StatusFinished
+	s.runs[2].Status = models.StatusFinished
+	assert.Nil(s.T(), s.runFixtures.UpdateRun(context.Background(), s.runs[2]))
+}
+
+func (s *GetRunsActiveTestSuite) Test_Ok() {
+	defer func() {
+		assert.Nil(s.T(), s.runFixtures.UnloadFixtures())
+	}()
+	tests := []struct {
+		name         string
+		wantRunCount int
+	}{
+		{
+			name:         "GetActiveRunsWhenPresent",
+			wantRunCount: 2,
+		},
+		{
+			name:         "GetActiveRunsWhenNotPresent",
+			wantRunCount: 0,
+		},
+	}
+	for _, tt := range tests {
+		s.T().Run(tt.name, func(T *testing.T) {
+			if tt.wantRunCount == 0 {
```

Comment:
it is hard to understand what is going on here when you read this moment first time. Can you please remove this test at all? I can't see any reason to have it like I mentioned before.

Reply:
this is the test for empty db table -- so it needs a different setup. I don't think we should remove it, seems like a valid test case. Either we move setup into the test for other cases or clean up in this case -- but I think it helps readability to leave setup outside the test function as much as possible.

Reply:
added a comment, hopefully that helps? let me know

Reply:
removing